Feature/loading scripts

andkaha opened the pull request:
New loading scripts added.

These scripts load the CSV data into a `data` table and uses SQL to populate the tables of the main database from there.  A database dump file of the loaded tables is then created.

See also the `README` file.

andkaha committed:
Initial check-in of loading scripts

TODO: Write README

andkaha committed:
Pick out herd names from data

andkaha committed:
Get herd names for Gotlandskanin from aux CSV file

andkaha committed:
Protect against dupes

andkaha committed:
Add aux herd data with two UPDATEs instead

andkaha committed:
Load aux herd data totally separately

andkaha committed:
Change database dump filename

andkaha committed:
Spelling

andkaha committed:
Add herd names later, as for Gotland

andkaha committed:
Don't clobber existing herd names

andkaha committed:
Fix dupe ".dump" in dump output filename

andkaha committed:
Add README

andkaha committed:
Say something about conversion to CSV and how to run

andkaha committed:
Remove old TRUNCATE statements

andkaha committed:
Comments

andkaha committed:
Don't dump the "colour" table at thet end (as it's not loaded by these scripts)

jhagberg reviewed (changes requested):
We get the data as XLS but this takes CSV can we incorporate so we can load the original file directly instead of using some external thing to translate the xls file to CSV?

jhagberg commented:
in2csv is a command in csvkit can that be used instead of the Google step?

andkaha commented:
I will investigate the `in2csv` utility (which I haven't tried using before).

jhagberg commented on scripts/load-gotland.sh:
```diff
@@ -0,0 +1,170 @@
+	--tables data2 \
+	--overwrite \
+	--skip-lines 5 \
+	--insert herd-registry-gotland.csv
```
There is no information in the Readme about this file

andkaha replied:
Will fix that later in the afternoon!

andkaha replied:
This is now addressed in `README`, I hope.

jhagberg replied:
Using in2csv --skip-lines 5 worked for that file xlsx file I think! Should we specify the name on the cmd line also?

andkaha replied:
This (using `--skip-lines 5`) is what I did in the update version.

Since the auxiliary file is not expected to change very often, I opted for hard-coding its name in the script.  This is in contrast to the other data files. These may change often.

Also, I kinda hope these loading script should be of _temorary_ use until such time that data entry happens via the web site.

andkaha committed:
Use original Excel files instead of CSV

andkaha committed:
Mention herd-registry-gotland.xlsx more explicitly

jhagberg commented:
When running the script I get "expected to find ../app/config.ini" so I guess you need to build the docker images first. I did this but probably in another branch

andkaha committed:
Verify correct filename suffix on files named on command line

andkaha committed:
Work off CSV files

Convert the Excel sheets to CSV if given XLSX files and let the two
scripts use CSV data instead of the original files.  This ensures that
we have a CSV version of the data available, which I need to have to
aid further development.

andkaha committed:
Comment fix

andkaha committed:
Move connection string creation into load.sh

This makes load.sh use all the various configuration bit from config.ini
when creating the necessary environment variables etc. so that you're
able to connect to a remote database and not just to a local socket.

andkaha committed:
Tweak dump file name

Commit:
Update init_db.sh

Install pydigree requirments first

andkaha committed:
Merge branch 'feature/loading-scripts' of github.com:NBISweden/theherdbook into feature/loading-scripts

andkaha committed:
Nicify

Separate $year and $column.  For the tracking data, these will be the
same, but for the weight and bodyfat data, they will not.

andkaha committed:
Load weight data

andkaha committed:
Load bodyfat data

andkaha committed:
Use command line options for input filenames

Usage:
    ./load.sh   \
    	-g kanindata-gotland.xlsx -G herd-registry-gotland.xlsx \
    	-m kanindata-mellerud.xlsx

Options:

    -g file     Load Gotland rabbit data from "file"
    -G file     Load Gotland herd data from "file"
    -m file     Load Mellerud rabbit data from "file"

    -h          Show this help text

andkaha committed:
Update README

andkaha committed:
Put redirection operators where most devs expect to find them

andkaha commented:
I merged this as all review issues have been addressed, and since no further serious issues have surfaced.  Further changes to the loading script code will be addressed in future PRs.
